srv/methods.go

```diff
@@ -54,6 +58,121 @@ func (s *APIServer) jrpcMethods() jrpc.MethodMap {
 
 }
 
+type ResultGlobalRichList struct {
+	Address string `json:"address"`
+	Equiv   uint64 `json:"pusd"`
+}
+
+func (s *APIServer) getGlobalRichList(data json.RawMessage) interface{} {
+	params := ParamsGetGlobalRichList{}
+	_, _, err := validate(data, &params)
+	if err != nil {
+		return err
+	}
+
+	if params.Count == 0 {
+		params.Count = 100
+	}
+
+	height := s.Node.GetCurrentSync()
+	rates, err := s.Node.Pegnet.SelectRates(nil, height)
+	if err != nil {
+		return err
+	}
+
+	rich, err := s.Node.Pegnet.SelectAllBalances()
+	if err != nil {
+		return err
+	}
+
+	var res []ResultGlobalRichList
+	for _, r := range rich {
+		var usd uint64
+
+		for i := fat2.PTicker(1); i < fat2.PTickerMax; i++ {
+			if r.Balances[i] == 0 {
+				continue
+			}
+			c, err := conversions.Convert(int64(r.Balances[i]), rates[i], rates[fat2.PTickerUSD])
+			if err != nil {
+				return err
+			}
+
+			usd += uint64(c)
+		}
+
+		if usd == 0 {
+			continue
+		}
+
+		var entry ResultGlobalRichList
+		entry.Address = r.Address.String()
+		entry.Equiv = usd
+
+		res = append(res, entry)
+	}
+
+	sort.Slice(res, func(i, j int) bool {
+		return res[i].Equiv > res[j].Equiv
+	})
+
+	if len(res) > params.Count {
+		res = res[:params.Count]
+	}
+
+	if len(res) == 0 {
+		return []ResultGlobalRichList{}
+	}
+	return res
+}
+
+type ResultGetRichList struct {
+	Address string `json:"address"`
+	Amount  uint64 `json:"amount"`
+	Equiv   uint64 `json:"pusd"`
+}
+
+func (s *APIServer) getRichList(data json.RawMessage) interface{} {
+	params := ParamsGetRichList{}
+	_, _, err := validate(data, &params)
+	if err != nil {
+		return err
+	}
+
+	if params.Count == 0 {
+		params.Count = 100
+	}
+
+	height := s.Node.GetCurrentSync()
+	rates, err := s.Node.Pegnet.SelectRates(nil, height)
+	if err != nil {
+		return err
+	}
+
+	ticker := fat2.StringToTicker(params.Asset) // already validated
+
+	rich, err := s.Node.Pegnet.SelectRichList(ticker, params.Count)
+	if err != nil {
+		return err
+	}
+
+	var res []ResultGetRichList
```

Comment:
If there is no asset owners, so the return is a nil or empty array, this will return a nil. 
We should replace `var res []ResultGetRichList` with `res := make([]ResultGetRichList, 0)` so we get an empty list vs a null.

Reply:
good call. i originally had a different return structure and didn't fix this when i changed it